### PR TITLE
reanme CONFIG_USER_ENTRYPOINT  to CONFIG_INIT_ENTRYPOINT

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

1. modify the name of USERMAIN_ENTRY
2. use exec_spawn instead of exec

case of the one choice of "Initialization Task" is INIT_ENTRYPOINT, but the entry point is USER_ENTRYPOINT,
this is not for suitable.
So, I modify the choice to INIT_ENTRY, and the entry point is INIT_ENTRYPOINT.

At last:
choice INIT_ENTRY --> INIT_ENTRYPOINT  (INIT_ENTRYXX)
choice INIT_FILE     --> INIT_FILEAPTH        (INIT_FILEXX)

and:
INIT_STACKSIZE & INIT_PRIORITY for all choices

## Impact

usermain entry
serial

## Testing

vela

### CODE REVIEW

File: drivers/serial/Kconfig [hunk 1]
```diff
@@ -173,6 +173,69 @@ config SERIAL_TERMIOS
 		If this is not defined, then the terminal settings (baud, parity, etc).
 		are not configurable at runtime; serial streams cannot be flushed, etc..
 
+config TTY_LAUNCH
+	bool "Enable feature TTY launch program"
+	default n
+	---help---
+		If select this, then user can lanunch a program with a special input.
```
Comment (acassis) [inline drivers/serial/Kconfig:180]: lanunch -> launch

**Comment by acassis** [comment]
@GUIDINGLI could you please provide more details of these modification? I noticed that CONFIG_INIT_ENTRYPOINT was renamed to CONFIG_INIT_ENTRY but the old name still referenced in many places. I suggest we test it more before incorporating it on mainline.

Also the TTY_LAUNCH purpose is not too clear because currently we already have mechanisms to start NuttX using other TTY inputs (see CONFIG_NSH_ALTCONDEV). To avoid what Tim just called "Obscure Menuconfig" I suggest you to create a documentation about this feature to let people know how to use it.

**Review by acassis — CHANGES_REQUESTED** [review]
Please create a Documentation/ entry about this feature

File: Documentation/applications/nsh/installation.rst [hunk 2]
```diff
@@ -14,11 +14,11 @@ increasing difficulty:
   #. You replace the sample code at ``apps/examples/nsh/nsh_main.c`` with
      whatever start-up logic that you want. NSH is a library at
      ``apps/nshlib``. ``apps.examples/nsh`` is just a tiny, example
-     start-up function (``CONFIG_USER_ENTRYPOINT``\ ()) that that runs
+     start-up function (``CONFIG_INIT_ENTRYPOINT``\ ()) that that runs
```
Comment (gustavonihei) [inline Documentation/applications/nsh/installation.rst:17]:
```suggestion
     start-up function (``CONFIG_INIT_ENTRYPOINT``\ ()) that runs
```
Could you please remove this duplicated "that"? Thanks.

File: drivers/serial/serial.c [hunk 3]
```diff
@@ -1589,6 +1605,58 @@ static int uart_poll(FAR struct file *filep,
...
+  if (!strcmp(tcb->name, CONFIG_TTY_LAUNCH_ENTRYNAME))
+#else
+  if (!strcmp(tcb->name, CONFIG_TTY_LAUNCH_FILEPATH))
+#endif
+    {
+      *(int *)arg = 1;
+    }
+}
+
+static void uart_lanuch_worker(void *arg)
```
Comment (acassis) [inline drivers/serial/serial.c:1625]: uart_lanuch_worker -> uart_launch_worker

File: drivers/serial/serial.c [hunk 4]
```diff
@@ -1589,6 +1605,58 @@ static int uart_poll(FAR struct file *filep,
...
+  {
+    CONFIG_TTY_LAUNCH_ARGS,
+    NULL,
+  };
+#else
+  FAR char *const *argv = NULL;
+#endif
+  int found = 0;
+
+  nxsched_foreach(uart_lanuch_foreach, &found);
```
Comment (acassis) [inline drivers/serial/serial.c:1638]: uart_lanuch_foreach -> uart_launch_foreach

File: include/nuttx/serial/serial.h [hunk 5]
```diff
@@ -496,6 +496,19 @@ void uart_recvchars_done(FAR uart_dev_t *dev);
 
 void uart_reset_sem(FAR uart_dev_t *dev);
 
+/****************************************************************************
+ * Name: uart_lanch
+ *
+ * Description:
+ *   This function is called when user want lanch a new program by
```
Comment (acassis) [inline include/nuttx/serial/serial.h:503]: lanch -> launch

**Review by davids5 — COMMENTED** [review]
Please justify the value of name change. What value does it have?  If others think it should come in, it should not be a separate PR.

File: drivers/serial/Kconfig (hunk 1 above)
Comment (GUIDINGLI) [reply drivers/serial/Kconfig]: next patch

**Comment by GUIDINGLI** [comment]
> @GUIDINGLI could you please provide more details of these modification? I noticed that CONFIG_INIT_ENTRYPOINT was renamed to CONFIG_INIT_ENTRY but the old name still referenced in many places. I suggest we test it more before incorporating it on mainline.
> 
> Also the TTY_LAUNCH purpose is not too clear because currently we already have mechanisms to start NuttX using other TTY inputs (see CONFIG_NSH_ALTCONDEV). To avoid what Tim just called "Obscure Configurations" I suggest you to create a documentation about this feature to let people know how to use it.

TTY_LAUNCH will be next RP, others please see Summary

**Comment by GUIDINGLI** [comment]
> Please justify the value of name change. What value does it have? If others think it should come in, it should not be a separate PR.

@davids5 done, please see Summary

**Review by acassis — APPROVED** [review]

**Comment by yamt** [comment]
@GUIDINGLI @acassis @xiaoxiang781216 
what's the motivation of the config rename?
i read the description but i failed to understand.
this broke the existing configs without any obvious benefit.
i tend to think it should be reverted.

**Comment by pkarashchenko** [comment]
Yeah. Seems to be a "breaking change" without any marker

**Comment by xiaoxiang781216** [comment]
It already highlight in the release note:
https://cwiki.apache.org/confluence/display/NUTTX/NuttX+10.3

**Comment by yamt** [comment]
good to hear it's mentioned in the release notes.
i'm still not sure what was the motivation of this rename though.

**Comment by xiaoxiang781216** [comment]
To unify the prefix to INIT_:
https://github.com/apache/incubator-nuttx/pull/5065/files#diff-285d196ad580d55fdaef3fbe54702ee3ddeed0d93cba52a5441d1866e0031726R302-R430

**Comment by yamt** [comment]
> To unify the prefix to INIT_: https://github.com/apache/incubator-nuttx/pull/5065/files#diff-285d196ad580d55fdaef3fbe54702ee3ddeed0d93cba52a5441d1866e0031726R302-R430

ok.
i feel it doesn't warrant a breaking change.
but thank you for explanation anyway.

**Pull request merged**